worksheet_B/Mandelbrot.cpp

```diff
@@ -15,36 +15,74 @@ int main()
 	const double minX = -2, maxX = 1, minY = -1.5, maxY = 1.5;
 
 	// Generate the image
+	float imageWidth = image.width();
+	float imageHeight = image.height();
+	int R =0, G = 0, B = 0; //initialising colours
+
 	for (int pixelY = 0; pixelY < image.height(); pixelY++)
 	{
-		// TODO: Map the y coordinate into the range minY to maxY
-		//double y0 =
+		// Map the y coordinate into the range minY to maxY
+		double y0 = ((pixelY / imageHeight) * (maxY - minY) + minY);
 
 		for (int pixelX = 0; pixelX < image.width(); pixelX++)
 		{
-			// TODO: Map the x coordinate into the range minX to maxX
-			//double x0 =
+			// Map the x coordinate into the range minX to maxX
+			double x0 = ((pixelX / imageWidth) * (maxX - minX) + minY);
 
-			// TODO: implement the algorithm to colour a single pixel (x0, y0) of the Mandelbrot set fractal
+			// Algorithm to colour a single pixel (x0, y0) of the Mandelbrot set fractal
 			// The code below simply fills the screen with random pixels
+			double x = 0.0;
+			double y = 0.0;
+
+			double iteration = 0;
+			int maxIteration = 200;
+
+			while ((x*x + y*y < 2 * 2) && (iteration < maxIteration))
+			{
+				double xtemp = x*x - y*y + x0;
+				y = 2 * x*y + y0;
+				x = xtemp;
+				++iteration;
+			}
+
+			// Break down of colour formula
+			double i = iteration; // Used in the formula
+			double loopPoint = 10; // Initial number of iterations before looping
+			double loopMultiplier = 1.2; // Increase in the amount of iterations needed before looping
+
+			if (iteration > loopPoint)
+			// loops colours and then increases loop size each time
+			{
+				i = iteration - loopPoint;
+				loopPoint *= loopMultiplier;
+			}
+
+
+			double firstRatio = (i) / (i + 1); // Ration of I(n) : I(n + 1)
+			double secondRatio = (i + 1) / (i + 2); // Ratio of I(n + 1) : I(n + 2)
+			double ratioDifference = secondRatio - firstRatio; // produces decimal values from 0.167, tending towards 0 when iteration = infinity
+			double contrast = 6; // 0 = no contrast, 6 = max contrast (values higher than 6 results in rgb values above 255, thus looping)
```

Comment:
Interesting effect, but would benefit from a comment explaining at a high level what it does.

More importantly, the worksheet asked you to use an HSV-to-RGB algorithm to compute the colours, yet you have not!